filecoin-hashers/src/poseidon_types.rs

```diff
@@ -26,55 +30,71 @@ lazy_static! {
     pub static ref POSEIDON_CONSTANTS_11: PoseidonConstants::<Fr, U11> = PoseidonConstants::new();
     pub static ref POSEIDON_MD_CONSTANTS: PoseidonConstants::<Fr, PoseidonMDArity> =
         PoseidonConstants::new();
-}
+    pub static ref POSEIDON_CONSTANTS_2_PALLAS: PoseidonConstants::<Fp, U2> =
+        PoseidonConstants::new();
+    pub static ref POSEIDON_CONSTANTS_4_PALLAS: PoseidonConstants::<Fp, U4> =
+        PoseidonConstants::new();
+    pub static ref POSEIDON_CONSTANTS_8_PALLAS: PoseidonConstants::<Fp, U8> =
+        PoseidonConstants::new();
+    pub static ref POSEIDON_CONSTANTS_11_PALLAS: PoseidonConstants::<Fp, U11> =
+        PoseidonConstants::new();
+    pub static ref POSEIDON_MD_CONSTANTS_PALLAS: PoseidonConstants::<Fp, PoseidonMDArity> =
+        PoseidonConstants::new();
+    pub static ref POSEIDON_CONSTANTS_2_VESTA: PoseidonConstants::<Fq, U2> =
+        PoseidonConstants::new();
+    pub static ref POSEIDON_CONSTANTS_4_VESTA: PoseidonConstants::<Fq, U4> =
+        PoseidonConstants::new();
+    pub static ref POSEIDON_CONSTANTS_8_VESTA: PoseidonConstants::<Fq, U8> =
+        PoseidonConstants::new();
+    pub static ref POSEIDON_CONSTANTS_11_VESTA: PoseidonConstants::<Fq, U11> =
+        PoseidonConstants::new();
+    pub static ref POSEIDON_MD_CONSTANTS_VESTA: PoseidonConstants::<Fq, PoseidonMDArity> =
+        PoseidonConstants::new();
+    pub static ref POSEIDON_CONSTANTS: ShareMap = {
+        let mut tm = ShareMap::custom();
 
-pub trait PoseidonArity: Arity<Fr> + Send + Sync + Clone + Debug {
-    #[allow(non_snake_case)]
-    fn PARAMETERS() -> &'static PoseidonConstants<Fr, Self>;
```

Comment:
Not: remove `PoseidonArity::PARAMETERS` in favor of `POSEIDON_CONSTANTS` lookup table.